### PR TITLE
Added tests for Out Variable Declarations in new expression contexts.

### PR DESCRIPTION
**Pull request by AlekseyTs**
@dotnet/roslyn-compiler Please review. This is the last chunk of tests ported from pattern-matching.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review. This is the last chunk of tests ported from pattern-matching.

File: src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs [hunk 1]
```diff
@@ -344,17 +345,35 @@ private static bool ReadOutside(ExpressionSyntax dataFlowParent, IdentifierNameS
...
+            {
+                return false;
+            }
+
+            var containingStatement = decl.Ancestors().OfType<StatementSyntax>().FirstOrDefault();
+            var containingReturnOrThrow = containingStatement as ReturnStatementSyntax ?? (StatementSyntax)(containingStatement as ThrowStatementSyntax);
+
+            MethodDeclarationSyntax methodDeclParent;
+
+            if (containingReturnOrThrow != null && decl.Identifier.ValueText == "x1" &&
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: Why do we have a constant of value `"x1"` in the code base?  That seems off.

Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: Nevermind, just realized this is in the actual test code.

---
In reply to: [67759021](https://github.com/dotnet/roslyn/pull/12077#discussion_r67759021) [](ancestors = 67759021)

**Comment by jaredpar** [comment]
:shipit:

**Comment by jcouv** [comment]
LGTM. Thanks

**Pull request merged**